lib/std/os/uefi/protocol/file.zig

```diff
@@ -40,95 +57,161 @@ pub const File = extern struct {
         return .{ .context = self };
     }
 
-    pub fn open(self: *const File, new_handle: **const File, file_name: [*:0]const u16, open_mode: u64, attributes: u64) Status {
-        return self._open(self, new_handle, file_name, open_mode, attributes);
-    }
-
-    pub fn close(self: *const File) Status {
-        return self._close(self);
+    pub fn open(
+        self: *const File,
+        file_name: [*:0]const u16,
+        open_mode: u64,
+        attributes: u64,
+    ) !*File {
+        var new: *File = undefined;
+        switch (self._open(self, &new, file_name, open_mode, attributes)) {
+            .success => return new,
+            .not_found => return Error.NotFound,
+            .no_media => return Error.NoMedia,
+            .media_changed => return Error.MediaChanged,
+            .device_error => return Error.DeviceError,
+            .volume_corrupted => return Error.VolumeCorrupted,
+            .write_protected => return Error.WriteProtected,
+            .access_denied => return Error.AccessDenied,
+            .out_of_resources => return Error.OutOfResources,
+            .volume_full => return Error.VolumeFull,
+            .invalid_parameter => return Error.InvalidParameter,
+            else => |err| return uefi.unexpectedError(err),
+        }
     }
 
-    pub fn delete(self: *const File) Status {
-        return self._delete(self);
+    pub fn close(self: *File) void {
+        switch (self._close(self)) {
+            .success => {},
+            else => |err| uefi.unexpectedError(err),
+        }
     }
 
-    pub fn read(self: *const File, buffer_size: *usize, buffer: [*]u8) Status {
-        return self._read(self, buffer_size, buffer);
+    /// Delete the file.
+    ///
+    /// Returns true if the file was deleted, false if the file was not deleted, which is a warning
+    /// according to the UEFI specification.
+    pub fn delete(self: *File) bool {
+        switch (self._delete(self)) {
+            .success => return true,
+            .warn_delete_failure => return false,
```

Comment:
That seems like a good solution. Once we find an API that can return both a value and a warning it'll get awkward though.